lib.py

```diff
@@ -0,0 +1,305 @@
+import time
+from telebot import types
+import yaml
+from pathlib import Path
+import logging
+from itertools import zip_longest
+import random
+logger = logging.getLogger(__name__)
+import threading
+lock = threading.Lock()
+
+
+#TODO
+# ----1) сделать отдельный файл с объявлениями ф-ий (
+# 2) PyTest (или ст питоновский юнит тест import unittest (test.py)
+# 3) ПРОБЛЕМА прога зависит от того, когда я её запустил - нужен график выполнения функции round_create
+# ----4) JSON либо yaml - под хранение БД - пока через txt до прода доведу лучше
+# ----5) сделать write_active_participants()
+# ----6) что если один чел в конце останется?
+# 7) если YAML пустой на моменте вызова ф-ии read(), то всё падает
+# ----8) file threads safety! Mutex?
+# 9) в main() процесс не идёт дальше bot.poling
+# 10) create random_pairing()
+
+
+def read_users():
+    """
+    read users from db/users.yaml
+    """
+    # users = yaml.safe_load(str(Path('db', 'users.yaml')))
+    with lock:
+        with open('db/users.yaml') as f:
+            users = yaml.safe_load(f)
+    # logger.info(users)
+    print(type(users))
+    print(users)
+    return users
+
+
+def add_user(user: dict):
+    """
+    read users from db/users.yaml,
+    add new_user in buffer dic,
+    rewrite the BD.
+    """
+    users = read_users()
+    users.update(user)
+    print(users)  # just checking
+    write_users(users)
+
+
+def remove_user(user):
+    """
+    reads current BD,
+    deleting delUser from buffer dict,
+    rewriting the BD.
+    """
+    users = read_users()
+    del users[user]
+    print(users)  # just checking
+    write_users(users)
+
+
+def write_users(users):
+    """
+    write to db/users.yaml
+    """
+    with lock:
```

Comment:
We need another mutex for preventing a race condition, where one user could prevent another from registering/deleting herself. (comment 5/5)
```suggestion
    with file_lock:
```